api/authenticator.py

```diff
@@ -921,12 +921,31 @@ def create_authentication_document(self):
             links=links
         ).to_dict(self._db)
 
-        # Add the library's color scheme, if it has one.
+        # Add the library's mobile color scheme, if it has one.
         description = ConfigurationSetting.for_library(
             Configuration.COLOR_SCHEME, library).value
         if description:
             doc['color_scheme'] = description
 
+        # Add the library's web colors, if it has any.
+        background = ConfigurationSetting.for_library(
+            Configuration.WEB_BACKGROUND_COLOR, library).value
+        foreground = ConfigurationSetting.for_library(
+            Configuration.WEB_FOREGROUND_COLOR, library).value
+        if background or foreground:
+            doc["web_colors"] = dict(background=background, foreground=foreground)
+
+        # Add the library's web header links, if it has any.
+        header_urls = ConfigurationSetting.for_library(
+            Configuration.WEB_HEADER_LINKS, library).json_value
+        if header_urls:
+            header_labels = ConfigurationSetting.for_library(
+                Configuration.WEB_HEADER_LABELS, library).json_value
+            header_links = []
+            for (url, label) in zip(header_urls, header_labels):
+                header_links.append(dict(href=url, title=label));
+            doc["web_header_links"] = header_links
+
```

Comment:
These things should be documented as [extensions](https://github.com/NYPL-Simplified/Simplified/wiki/Authentication-For-OPDS-Extensions) to Authentication For OPDS.

For `web_colors` I think it would be good enough to call it `web_color_scheme` to fit with `color_scheme`.

For `web_header_links` it's more difficult. I can think of a couple solutions:

* Leave `web_header_links` alone. Define it as being the same as `links` but for links specific to the 'web header' role. The problem with this is that `links` requires a `rel` for each link. So we'd either define `rel` as optional for `web_header_links` or make up a fake rel for it.

* Incorporate `web_header_links` into `links` somehow, with "web header" as the `role` (which would be different from a `rel` -- `rel` describes the relationship between the two pages, `role` describes the purpose of the link within the document.)

* Create a separate object called `web` and reuse the standard names `links` and `color_scheme` inside that object, possibly defining them differently.

Reply:
This doesn't absolutely need to be resolved in this branch but it would be best to reach an agreement before this is deployed on servers we don't control.

Reply:
I'd like to resolve this now, I was waiting for your feedback before I updated the wiki.

I kind of like the second solution. Which one do you prefer?

Reply:
I also prefer the second solution. The problem is coming up with a `rel` -- we'd need to either let the user specify an appropriate `rel` for each link (which is kind of silly because in a web context it doesn't matter), or make up a generic fake `rel`, or have 'web header link' be the rel, or leave it blank and serve invalid Authentication For OPDS documents.

Reply:
Letting 'web header link' be the `rel` is the simplest solution but it twists what a `rel` is for.

Reply:
What if we used "related" as the `rel` for these and kept "web header link" as the `role`?

Reply:
I like that, and I think the role should be `navigation` based on https://www.w3.org/1999/xhtml/vocab.

Reply:
That does make sense. We already get footer links from the feed (about, TOS, etc).

Reply:
Yeah, that looks a lot better.